api/src/client/ApiFormationHelpers.ts

```diff
@@ -0,0 +1,546 @@
+  };
+  CompanyProfit: "Profit" | "NonProfit";
+  RegisteredAgent: {
+    Id: string | undefined; // 7 max, must be valid NJ registered agent number
+    Email: string | undefined; //50 max, must be email address
+    Name: string | undefined; // 50 max, required if no ID
+    Location: ApiLocation | undefined; // required if no ID
+  };
+  MemberAttestation: boolean | undefined;
+  Incorporators: MembersObjects[] | undefined;
+  Members: MembersObjects[] | undefined;
+  Signers: Array<{
+    //This can be a list/array of items. Maximum 10
+    Name: string; // max 50
+    Title: SignerTitle;
+    Signed: boolean;
+  }>;
+  ContactFirstName: string; //Contact person for filing
+  ContactLastName: string;
+  ContactPhoneNumber: string;
+}
+
+export interface ApiSubmission {
+  Account: string; //API User Account id
+  Key: string; //API Account Key
+  ReturnUrl: string;
+  FailureReturnUrl: string;
+  Payer: {
+    FirstName?: string; // string (combined with LastName total 49), if not set will use ContactFirstName
+    LastName?: string; // string (combined with FirstName total 49), if not set will use ContactLastName
+    CompanyName: string;
+    Address1: string;
+    Address2: string;
+    City: string;
+    StateAbbreviation?: StateShortCodes; // 2 chars
+    ZipCode?: string; // 5 chars
+    PhoneNumber?: string; // 12 chars, if not set ContactPhoneNumber will be used
+    Email?: string;
+  };
+  Formation: Formation;
+  ForeignGoodStandingFile?: {
+    Extension: string;
+    Content: string;
+  };
+}
+
+export type MembersObjects = {
+  Name: string; // 50 max
+  Location: ApiLocation;
+};
+
+export type AdditionalProvision = {
+  Provision: string; // max 400 chars
+};
+
+export type ApiLocation = {
+  Address1: string; // 35 char max, Can not be left blank, if any other address field has data
+  Address2: string; // max 35 char
+  City: string; // 30 char max, Can not be left blank, if any other address field has data
+  State: StateNames | undefined; // required if country = US. Must be full state name. Ex: Alabama
+  Province?: string;
+  Zipcode: string; // max 5 if country=US, 11 otherwise
+  Country: CountriesShortCodes | undefined; //alpha-2 iban code
+};
+
+export type BusinessType =
+  | "DomesticLimitedLiabilityCompany"
+  | "DomesticLimitedLiabilityPartnership"
+  | "DomesticForProfitCorporation"
+  | "DomesticNonProfitCorporation"
+  | "DomesticNonProfitVeteranCorporation"
+  | "ForeignNonProfitCorporation"
+  | "DomesticLimitedPartnership"
+  | "ForeignForProfitCorporation"
+  | "ForeignLimitedLiabilityCompany"
+  | "ForeignLimitedPartnership"
+  | "ForeignLimitedLiabilityPartnership";
+
+export type FormationFields = {
+  businessType: BusinessType;
+  shortDescription: "LLC" | "LLP" | "LP" | "DP" | "NP" | "NV" | "NF" | "LF" | "LFC" | "FLC" | "FLP" | "FR";
+};
+
+export const BusinessTypeMap: Record<FormationLegalType | "veteran-nonprofit", FormationFields> = {
```

Comment:
[sand] Some of these functions and variables are only being used in this file. I'm wondering if they got exported automatically by the IDE when you restructured the files.
```suggestion
const BusinessTypeMap: Record<FormationLegalType | "veteran-nonprofit", FormationFields> = {
```

Reply:
done